x-pack/plugins/apm/server/routes/agent_keys/invalidate_agent_key.ts

```diff
@@ -9,17 +9,20 @@ import { ApmPluginRequestHandlerContext } from '../typings';
 export async function invalidateAgentKey({
   context,
   id,
+  isAdmin,
 }: {
   context: ApmPluginRequestHandlerContext;
   id: string;
+  isAdmin: boolean;
 }) {
+  const body: { ids: string[]; owner?: boolean } = { ids: [id] };
+  if (!isAdmin) {
+    body.owner = true;
+  }
   const { invalidated_api_keys: invalidatedAgentKeys } =
     await context.core.elasticsearch.client.asCurrentUser.security.invalidateApiKey(
       {
-        body: {
-          ids: [id],
-          owner: true,
-        },
+        body,
```

Comment:
What about
```suggestion
        body: { ids: [id], owner: isAdmin };
```

Reply:
If isAdmin is false, owner should be true

Reply:
The default value for `owner` is false, we need to set it to true when user doesn't have privileges so if it has role `manage_own_api_key` it can be invalidated by the owner

Reply:
Isn't it just this, then?

```suggestion
        body: { ids: [id], owner: !isAdmin },
```

Reply:
This can work, but I think is misleading that the owner value depends on if it's admin or not, I rather keep it as "if it's not admin we add the owner flag and if it's admin we don't need to add it"
But if @sqren do you think it will improve the readability of the code I will change it

Reply:
>  but I think is misleading that the owner value depends on if it's admin or not

I think that the `owner` value indeed _does_ depends on whether the user is admin or not. If there are other things we should take into account we should do that.